### ISSUE TITLE
Exit code 1 with no errors in log or tests run when using workspace

### ISSUE THREAD

**Issue by zanizrules** [lyndsey-ferguson/fastlane-plugin-test_center#38]
Recently I tried to replace my scan function on a project with your multi_scan. I need to use the workspace rather than the project to cover my multiple projects which depend on each other... but multi_scan does not seem to work for this? Or am I missing something? I did something similar to your 'test_multiscan_workspace' example, except I had some extra parameters which I had previously passed into scan (xcodeargs, etc).

**Comment by lyndsey-ferguson** [comment]
Hello @zanizrules, `multi_scan` is intended to work in the way you describe. Can you provide a snippet of code demonstrating how you are calling `multi_scan` and provide the console log where it starts with this:
```
------------------------
--- Step: multi_scan ---
------------------------
```
until the end of the log?

**Comment by zanizrules** [comment]
Hi @lyndsey-ferguson :)

I've done some more debugging and have compared what my old scan function logs compared to what multi_scan is logging and one key difference is this:

Found in log when using scan: `clean test` 
Found in log when using multi_scan: `clean build-for-testing`

Now from the scan documentation I read that `build_for_testing` will skip tests all together which would explain why I don't see the emulator spawn up and tests start when using `multi_scan`.

I have tried passing in `build_for_testing: false` to my `multi_scan` but this had no effect. 

The Summary for scan table in the logs even has the following:
`| build_for_testing      | true                                                                        |`

The summary table for my old scan function did not even include a `build_for_testing ` so I can only assume this was due to it not being set/defaulting to false.

Here is how I am calling `multi_scan` 

`   multi_scan(
      workspace: './iOS/iOS.xcworkspace',
      scheme: 'ReRun',
      clean: true,
      skip_build: true,
      build_for_testing: false,
      xcargs: xcodeArgs,
      try_count: 3,
      batch_count: 2
    )
`

Also I am using fast lane version 2.86.2.

Any idea on how I can have this set back to false? does `multi_scan` turn this on for some reason?

**Comment by lyndsey-ferguson** [comment]
The way I have it structured, unless `:test_without_building` is given, `multi_scan` assumes that the test app is not built. I can see this as unnecessary if you provided an `:xctestrun` parameter that pointed to an existing pre-built app and test-runner app; however, even so, if you pass `clean: true`, it would wipe those directories out.

So, I need to confirm my understanding as I'm confused as how this works (and then I'll try it myself).

1. Your old way of calling scan was so, yes?
```ruby
scan( 
  workspace: './iOS/iOS.xcworkspace', 
  scheme: 'ReRun',
  clean: true,
  skip_build: true,
  xcargs: xcodeArgs
)
```

2. If so, `scan`, when not given `:test_without_building`, _will_ build and then test. As a result of that one call to `scan`. `multi_scan` separates the build and test steps so that the build will be done only once, and the test is performed `:try_count` times.

---

I still don't understand what error you are seeing yet. I would like to understand what your intent is by telling it _not_ to build, yet you want to clean the derived data (where the test app is) using the `:clean` option.

Can you provide the console log from when `multi_scan` starts until the error occurs?

**Comment by zanizrules** [comment]
My understanding was that `skip_build: true` would skip the debug build but the test build would still happen as needed. Whatever build the scan was defaulting to worked perfectly, but for some reason `test_without_building` is not working.

**Comment by lyndsey-ferguson** [comment]
You've thrown me a curve ball that I didn't expect 😄: I did not understand how `:skip_build` worked as I misunderstood it before and never really used it. Yes, `multi_scan` assumes that the build has to be performed and tries to build. 

In your first example, the `:build_for_testing` completes with an error and `multi_scan` stops. Why it would halt with an error, I cannot say as the logs are redacted.

In your second example, where you've reproduced the error with `scan`, perhaps you had to run it with `sudo` as it is expecting to write to a root owned path? The `scan` then runs and finds that the `test_result` bundle path already exists and that throws `xcodebuild` for a loop and it exits early.

---

In any case, `multi_scan` should respect `skip_build` and I'm looking into that.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.1.2](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/commit/8e17d4185c257ea6fcb7ba8e18cc03a59d5a8236)

**Comment by lyndsey-ferguson** [comment]
@zanizrules I believe that you are calling `scan` and `multi_scan` with `:result_bundle => true` as that is what I see in your output (not the example of the call). I have reproduced it and created Issue #40.

**Comment by lyndsey-ferguson** [comment]
The other people must not be using that option.

**Comment by lyndsey-ferguson** [comment]
Until I fix this, you can use the `:testrun_completed_block` to move the `test_result` bundle between test runs.

**Comment by lyndsey-ferguson** [comment]
It has been fixed in [v3.1.3](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.1.3)
